Sources/StreamChatUI/Utils/UIImageView+Extensions.swift

```diff
@@ -3,14 +3,18 @@
 //
 
 import Nuke
+import StreamChat
 import UIKit
 
 extension UIImageView {
     @discardableResult
-    func loadImage(
+    func loadImage<ExtraData: ExtraDataTypes>(
         from url: URL?,
         placeholder: UIImage? = nil,
         resizeAutomatically: Bool = true,
+        requestThumbnail: Bool = false,
```

Comment:
TBH this API now feels a bit complex. In the end, it all comes to the question - do you need the image in a specific size? If yes, you should try to get the correct URL and get it.

Maybe there's a way to simplify this?

Reply:
I was thinking about some enum with associated value like:
```
enum Resize {
    .resizeAfterDownload
    .requestThumbnail(size: CGSize?)
    .none
}
```
But you didn't like adding an extra option for ImageCrop and this would be the same. Adding extra public enum just for this. I would say it's a bit extensive, but not that complex. 

I'm open to suggestions on how to simplify this